core/state_transition.go

```diff
@@ -269,8 +269,18 @@ func (st *StateTransition) buyGas() error {
 }
 
 func (st *StateTransition) preCheck() error {
+	// Only check transactions that are not fake
+	if !st.msg.IsFake() {
+		// Make sure the sender is an EOA
+		code := st.state.GetCode(st.msg.From())
+		_, delegated := types.ParseDelegation(code)
+		if len(code) > 0 && !delegated {
+			return fmt.Errorf("%w: address %v, len(code): %d", ErrSenderNoEOA, st.msg.From().Hex(), len(code))
```

Comment:
Since this would be a failure scenario for enforced transactions, we should comment here why this is impossible in practice:

```suggestion
			// If the sender on L1 is a (delegated) EOA, then it must be a (delegated) EOA on L2, too.
			// If the sender on L1 is a contract, then we apply address aliasing.
			// The probability that the aliased address happens to be a smart contract on L2 is negligible.
			return fmt.Errorf("%w: address %v, len(code): %d", ErrSenderNoEOA, st.msg.From().Hex(), len(code))
```

Reply:
also checked this, and it seems that `appendEnforcedTransaction` only allows EOA to send transactions.

while also for delegated EOAs, they will pass this check, this check will only block normal contracts.

Reply:
We allow contracts now. See [this version](https://github.com/scroll-tech/scroll-contracts/blob/1a44cc4cd5bb29067051f59c09dd4407511f1a3a/src/L1/gateways/EnforcedTxGateway.sol#L108).

Reply:
@zimpha pls double check this.